make user fails

Hello @oterral 
as discussed before, there is an issue when trying to:
* ``make user``
* ``make testdebug``

```bash
PHANTOMJS_BIN="node_modules/.bin/phantomjs" node_modules/.bin/karma start test/karma-conf-debug.js;
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() ignores invisible layers and displays alert message for reprojected layers FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:430:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes a empty layer group FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:450:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes a layer group with layers FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:465:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes layers FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:485:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes legend FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:520:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes pdf legend FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:552:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes graticule FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:580:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() encodes overlays FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:622:25
PhantomJS 2.1.1 (Linux 0.0.0) ga_print_directive gaPrint on modern browsers when print is active #submit() uses multiprint FAILED
        expected undefined to equal true
        assert@test/lib/expect.js:102:16
        equal@test/lib/expect.js:216:16
        test/lib/expect.js:69:29
        expectConfig@test/specs/print/PrintDirective.spec.js:395:47
        test/specs/print/PrintDirective.spec.js:658:25
PhantomJS 2.1.1 (Linux 0.0.0): Executed 1146 of 1146 (9 FAILED) (19.493 secs / 7.888 secs)
Makefile:226: recipe for target 'testdebug' failed
make: *** [testdebug] Error 1

```

after the fix you did:

```diff
diff --git a/test/specs/print/PrintDirective.spec.js b/test/specs/print/PrintDirective.spec.js
index 11dbf78..9f0b9ca 100644
--- a/test/specs/print/PrintDirective.spec.js
+++ b/test/specs/print/PrintDirective.spec.js
@@ -2,7 +2,7 @@
 describe('ga_print_directive', function() {

   describe('gaPrint', function() {
-    var elt, scope, parentScope, $compile, $rootScope, map, $timeout, $httpBackend;
+    var elt, scope, parentScope, $compile, $rootScope, map, $timeout, $httpBackend, $translate;
     var $q, gaUrlUtils, $http, $window, gaPermalink, gaPrintLayer, gaAttribution, gaBrowserSniffer;
     var hrefUrl = 'http://localhost:8081/context.html?lang=en';
     var opt = {
@@ -194,6 +194,7 @@ describe('ga_print_directive', function() {
       $timeout = $injector.get('$timeout');
       $http = $injector.get('$http');
       $httpBackend = $injector.get('$httpBackend');
+      $translate = $injector.get('$translate');
       gaBrowserSniffer = $injector.get('gaBrowserSniffer');
       gaAttribution = $injector.get('gaAttribution');
       gaPrintLayer = $injector.get('gaPrintLayer');
@@ -382,7 +383,7 @@ describe('ga_print_directive', function() {
             expect(conf.units).to.be('m');
             expect(conf.rotation).to.be(0);
             expect(conf.app).to.be('config');
-            expect(conf.lang).to.be('en');
+            expect(conf.lang).to.be('de');
             expect(conf.dpi).to.be('150');
             expect(conf.qrcodeurl).to.be('http://foo.ch/qrcodegenerator?url=http%3A%2F%2Flocalhost%3A8081%2Fcontext.html%3Flang%3Den');
             expect(conf.pages[0].center).to.eql([10000, 10000]);
@@ -391,7 +392,7 @@ describe('ga_print_directive', function() {
             expect(conf.pages[0].scale).to.eql('500.0');
             expect(conf.pages[0].shortLink).to.eql('http://foo.ch');
             expect(conf.pages[0].rotation).to.eql(0);
-            expect(conf.pages[0].langen).to.be(true);
+            expect(conf.pages[0].langde).to.be(true);
           };

           beforeEach(function() {

```

``make user`` fails with:
```bash
$ make user
source rc_user && make all
Sourcing:
rc_user_ltclm
make[1]: Entering directory '/home/ltclm/mf-geoadmin3'
node_modules/.bin/eslint test/specs/ --fix

/home/ltclm/mf-geoadmin3/test/specs/print/PrintDirective.spec.js
  5:85  error  '$translate' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

Makefile:219: recipe for target 'linttest' failed
make[1]: *** [linttest] Error 1
make[1]: Leaving directory '/home/ltclm/mf-geoadmin3'
Makefile:184: recipe for target 'user' failed
make: *** [user] Error 2

```

--- a/src/components/featuretree/FeaturetreeDirective.js
+++ b/src/components/featuretree/FeaturetreeDirective.js
@@ -2,12 +2,14 @@ goog.provide('ga_featuretree_directive');
 
 goog.require('ga_layers_service');
 goog.require('ga_previewfeatures_service');
+goog.require('ga_translation_service');
 
 (function() {
 
   var module = angular.module('ga_featuretree_directive', [
     'ga_layers_service',
     'ga_previewfeatures_service',
+    'ga_translation_service',
     'pascalprecht.translate'
   ]);
 
@@ -20,13 +22,13 @@ goog.require('ga_previewfeatures_service');
 
   module.directive('gaFeaturetree',
       function($rootScope, $timeout, $http, $q, $translate, $sce, gaLayers,
-          gaPreviewFeatures) {
+          gaPreviewFeatures, gaLang) {
         var canceler = null;
         var timeoutPromise = null;
         var parser = new ol.format.GeoJSON();
 
         var getTranslatedLabel = function(obj) {
-          var possibleKey = 'label_' + $translate.use();
+          var possibleKey = 'label_' + gaLang.get();
           if (angular.isDefined(obj[possibleKey])) {
             return obj[possibleKey];
           } else {

--- a/src/components/print/PrintDirective.js
+++ b/src/components/print/PrintDirective.js
@@ -5,6 +5,7 @@ goog.require('ga_browsersniffer_service');
 goog.require('ga_layers_service');
 goog.require('ga_maputils_service');
 goog.require('ga_printlayer_service');
+goog.require('ga_translation_service');
 goog.require('ga_urlutils_service');
 
 (function() {
@@ -16,12 +17,13 @@ goog.require('ga_urlutils_service');
     'ga_attribution_service',
     'ga_maputils_service',
     'ga_layers_service',
-    'ga_urlutils_service'
+    'ga_urlutils_service',
+    'ga_translation_service'
   ]);
 
   module.controller('GaPrintDirectiveController', function($scope,
       $http, $q, $window, $translate, $timeout, gaLayers, gaMapUtils,
-      gaPermalink, gaBrowserSniffer, gaWaitCursor,
+      gaPermalink, gaBrowserSniffer, gaWaitCursor, gaLang,
       gaPrintLayer, gaAttribution, gaUrlUtils) {
     var pdfLegendString = '_big.pdf';
     var printRectangle;
@@ -250,7 +252,7 @@ goog.require('ga_urlutils_service');
       // http://mapfish.org/doc/print/protocol.html#print-pdf
       var view = $scope.map.getView();
       var proj = view.getProjection();
-      var lang = $translate.use();
+      var lang = gaLang.get();
       var defaultPage = {};
       defaultPage['lang' + lang] = true;
       var qrcodeUrl = $scope.options.qrcodeUrl +

--- a/src/components/query/QueryDirective.js
+++ b/src/components/query/QueryDirective.js
@@ -420,7 +420,7 @@ goog.require('ga_query_service');
     });
   });
 
-  module.directive('gaQuery', function($translate, gaBrowserSniffer, gaQuery,
+  module.directive('gaQuery', function(gaLang, gaBrowserSniffer, gaQuery,
       gaStyleFactory, gaMapUtils) {
     var dragBox;
     var dragBoxStyle = gaStyleFactory.getStyle('selectrectangle');
@@ -521,7 +521,7 @@ goog.require('ga_query_service');
             input.datetimepicker({
               pickDate: true,
               pickTime: false,
-              language: $translate.use()
+              language: gaLang.get()
             });
           }
         };

--- a/src/components/seo/SeoDirective.js
+++ b/src/components/seo/SeoDirective.js
@@ -2,16 +2,18 @@ goog.provide('ga_seo_directive');
 
 goog.require('ga_layers_service');
 goog.require('ga_seo_service');
+goog.require('ga_translation_service');
 (function() {
 
   var module = angular.module('ga_seo_directive', [
     'ga_layers_service',
     'ga_seo_service',
+    'ga_translation_service',
     'pascalprecht.translate'
   ]);
 
   module.directive('gaSeo',
-      function($sce, $timeout, $q, $http, $translate,
+      function($sce, $timeout, $q, $http, gaLang,
           gaSeo, gaLayers) {
         return {
           restrict: 'A',
@@ -152,7 +154,7 @@ goog.require('ga_seo_service');
                       replace('{Feature}', featureId);
                   $http.get(htmlUrl, {
                     params: {
-                      lang: $translate.use() // Left out other parameters as
+                      lang: gaLang.get() // Left out other parameters as
                       // they are not relevant for SEO
                       // (cadastralWbebMap Links)
                     }
@@ -292,7 +294,7 @@ goog.require('ga_seo_service');
 
                   $http.get(scope.options.identifyUrl, {
                     params: {
-                      lang: $translate.use(),
+                      lang: gaLang.get(),
                       geometryType: 'esriGeometryPoint',
                       geometryFormat: 'geojson',
                       geometry: east + ',' + north,

--- a/src/js/FeaturetreeController.js
+++ b/src/js/FeaturetreeController.js
@@ -1,14 +1,17 @@
 goog.provide('ga_featuretree_controller');
 
 goog.require('ga_print_service');
+goog.require('ga_translation_service');
+
 (function() {
 
   var module = angular.module('ga_featuretree_controller', [
-    'ga_print_service'
+    'ga_print_service',
+    'ga_translation_service'
   ]);
 
   module.controller('GaFeaturetreeController', function($http, $scope,
-      $timeout, $translate, $window, gaGlobalOptions, gaPrint) {
+      $timeout, gaLang, $window, gaGlobalOptions, gaPrint) {
 
     var featureTreeId = '#featuretree-popup';
     // List of layers using an extendHtmlPoup for the print instead of htmlPopup
@@ -75,7 +78,7 @@ goog.require('ga_print_service');
         winPrint = window.open('', 'printout');
       }
 
-      var lang = $translate.use();
+      var lang = gaLang.get();
       var printElementsLoaded = 0;
       var printLayers = [];
       printLayers['failure'] = {

--- a/src/js/MainController.js
+++ b/src/js/MainController.js
@@ -8,6 +8,7 @@ goog.require('ga_maputils_service');
 goog.require('ga_networkstatus_service');
 goog.require('ga_storage_service');
 goog.require('ga_topic_service');
+goog.require('ga_translation_service');
 goog.require('ga_window_service');
 
 (function() {
@@ -20,7 +21,8 @@ goog.require('ga_window_service');
     'ga_storage_service',
     'ga_background_service',
     'ga_topic_service',
-    'ga_window_service'
+    'ga_window_service',
+    'ga_translation_service'
   ]);
 
   /**
@@ -31,7 +33,7 @@ goog.require('ga_window_service');
       gaPermalinkFeaturesManager, gaPermalinkLayersManager, gaMapUtils,
       gaRealtimeLayersManager, gaNetworkStatus, gaPermalink, gaStorage,
       gaGlobalOptions, gaBackground, gaTime, gaLayers, gaTopic,
-      gaOpaqueLayersManager, gaMapLoad, gaWindow) {
+      gaOpaqueLayersManager, gaMapLoad, gaWindow, gaLang) {
 
     var createMap = function() {
       var toolbar = $('#zoomButtons')[0];
@@ -224,7 +226,7 @@ goog.require('ga_window_service');
     });
 
     $rootScope.$on('$translateChangeEnd', function() {
-      $scope.langId = $translate.use();
+      $scope.langId = gaLang.get();
     });
 
     $scope.time = gaTime.get();

--- a/test/specs/print/PrintDirective.spec.js
+++ b/test/specs/print/PrintDirective.spec.js
@@ -183,6 +183,11 @@ describe('ga_print_directive', function() {
         parent: window.parent,
         open: angular.noop
       });
+      $provide.value('gaLang', {
+        get: function() {
+          return 'en';
+        }
+      });
     };
 
     var injectServices = function($injector) {

